pkg/virt-handler/vm.go

```diff
@@ -344,16 +344,27 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 				domainInterfaceStatusByMac[domainInterfaceStatus.Mac] = domainInterfaceStatus
 			}
 
+			existingInterfacesSpecByName := map[string]v1.Interface{}
+			for _, existingInterfaceSpec := range vmi.Spec.Domain.Devices.Interfaces {
+				existingInterfacesSpecByName[existingInterfaceSpec.Name] = existingInterfaceSpec
+			}
+
 			// Iterate through all domain.Spec interfaces
 			for _, domainInterface := range domain.Spec.Devices.Interfaces {
 				interfaceMAC := domainInterface.MAC.MAC
 				var newInterface v1.VirtualMachineInstanceNetworkInterface
+				var isMasqueredeInterface bool = false
```

Comment:
Could we please hijack this PR to not update based on guest agent with both masqurade and slirp to implement  https://github.com/kubevirt/kubevirt/pull/3241.

You can rename it to `isForwardingBindingInterface`, so we keep that logic here and pick the reporting method based on a single variable. If you decide not to, fix the type in masquerade.

Reply:
@vatsalparekh what about slirp?